pkg/cli/options_test.go

```diff
@@ -34,6 +35,65 @@ import (
 )
 
 var _ = Describe("Discover external plugins", func() {
+	Context("validate plugins root path", func() {
+		var (
+			homePath   string = os.Getenv("HOME")
+			customPath string = "/usr/local/myplugins"
+			xdghome    string = os.Getenv("XDG_CONFIG_HOME")
+		)
+
+		It("based on the host os", func() {
+			err := os.Unsetenv("KB_PLUGINS_PATH")
+			Expect(err).To(BeNil())
+
+			plgPath, err := getPluginsRoot("darwin")
+			Expect(err).To(BeNil())
+			if xdghome != "" {
+				Expect(plgPath).To(Equal(fmt.Sprintf("%s/kubebuilder/plugins", xdghome)))
+				AddReportEntry(fmt.Sprintf("Detect XDG_CONFIG_HOME: %s, generated plugin path: %s", xdghome, plgPath), ReportEntryVisibilityAlways)
+			} else {
+				Expect(plgPath).To(Equal(fmt.Sprintf("%s/Library/Application Support/kubebuilder/plugins", homePath)))
+			}
+
+			plgPath, err = getPluginsRoot("linux")
+			Expect(err).To(BeNil())
+			if xdghome != "" {
+				Expect(plgPath).To(Equal(fmt.Sprintf("%s/kubebuilder/plugins", xdghome)))
+				AddReportEntry(fmt.Sprintf("Detect XDG_CONFIG_HOME: %s, generated plugin path: %s", xdghome, plgPath), ReportEntryVisibilityAlways)
+			} else {
+				Expect(plgPath).To(Equal(fmt.Sprintf("%s/.config/kubebuilder/plugins", homePath)))
+			}
+
+			plgPath, err = getPluginsRoot("random")
+			Expect(plgPath).To(Equal(""))
+			Expect(err).ToNot(BeNil())
+			Expect(err.Error()).To(ContainSubstring("host not supported"))
+		})
+
+		It("based on the user given path", func() {
```

Comment:
Nit: make the It descriptions more like full sentences. See the examples in https://onsi.github.io/ginkgo/